app/src/features/rules/components/RulesList/components/RulesTable/utils/rules.ts

```diff
@@ -46,23 +55,23 @@ export const getPinnedRules = (recordsMap: Record<string, RuleObj>) => {
 };
 
 // FIXME: Performance Improvements
-export const rulesToContentTableDataAdapter = (rules: RuleObj[]): RuleTableDataType[] => {
-  const ruleTableDataTypeMap: { [id: string]: RuleTableDataType } = {};
+export const recordsToContentTableDataAdapter = (records: StorageRecord[]): RuleTableDataType[] => {
+  const ruleTableDataTypeMap: { [id: StorageRecord["id"]]: RuleTableDataType } = {};
 
-  const groupRules = rules.filter((rule) => !!rule.groupId) as Rule[];
-  const nonGroupRules: RuleObj[] = rules
-    .filter((rule) => !rule.groupId)
-    .map((rule) => (rule.objectType === RuleObjType.GROUP ? { ...rule, children: [] } : rule));
+  const groupedRules = records.filter(isRuleWithGroupId);
```

Comment:
ungrouped rules have `groupId: ""`. `isRuleWithGroupId` will return true for all records